routers/authz_filter.go

```diff
@@ -35,20 +35,13 @@ type Object struct {
 }
 
 func getUsername(ctx *context.Context) (username string) {
-	defer func() {
-		if r := recover(); r != nil {
-			username, _ = getUsernameByClientIdSecret(ctx)
-		}
-	}()
-
-	username = ctx.Input.Session("username").(string)
-
-	if username == "" {
+	username, ok := ctx.Input.Session("username").(string)
```

Comment:
Why change here?

Reply:
Because this is causing a runtime panic when there is no username in the session leading to skipping `username = getUsernameByKeys(ctx)`. This part is only reachable when you log out so the username is an empty string.